From 99571304c828c2ec1e21a04d3cb79237167b1fce Mon Sep 17 00:00:00 2001
From: suiyan <suiyan@sunmmio.com>
Date: Mon, 29 Jun 2026 17:26:25 +0800
Subject: [PATCH 1/2] DADAO triple registration

Add dadao to ArchType enum, Triple.cpp switch statements,
and LLVM_ALL_TARGETS.
---
 llvm/CMakeLists.txt                     |  1 +
 llvm/include/llvm/TargetParser/Triple.h |  1 +
 llvm/lib/TargetParser/Triple.cpp        | 10 ++++++++++
 3 files changed, 12 insertions(+)

diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index fcbfed495383..8227843449c8 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -521,6 +521,7 @@ set(LLVM_ALL_TARGETS
   ARM
   AVR
   BPF
+  DADAO
   Hexagon
   Lanai
   LoongArch
diff --git a/llvm/include/llvm/TargetParser/Triple.h b/llvm/include/llvm/TargetParser/Triple.h
index 9c83abeeb3b1..3c96dde21d99 100644
--- a/llvm/include/llvm/TargetParser/Triple.h
+++ b/llvm/include/llvm/TargetParser/Triple.h
@@ -104,6 +104,7 @@ public:
     spirv64,     // SPIR-V with 64-bit pointers
     kalimba,     // Kalimba: generic kalimba
     shave,       // SHAVE: Movidius vector VLIW processors
+    dadao,       // DADAO: DADAO SimRISC 64-bit
     lanai,       // Lanai: Lanai 32-bit
     wasm32,      // WebAssembly with 32-bit pointers
     wasm64,      // WebAssembly with 64-bit pointers
diff --git a/llvm/lib/TargetParser/Triple.cpp b/llvm/lib/TargetParser/Triple.cpp
index a4f9dd42c0fe..048b3f1350e8 100644
--- a/llvm/lib/TargetParser/Triple.cpp
+++ b/llvm/lib/TargetParser/Triple.cpp
@@ -44,6 +44,7 @@ StringRef Triple::getArchTypeName(ArchType Kind) {
   case hsail:          return "hsail";
   case kalimba:        return "kalimba";
   case lanai:          return "lanai";
+  case dadao:          return "dadao";
   case loongarch32:    return "loongarch32";
   case loongarch64:    return "loongarch64";
   case m68k:           return "m68k";
@@ -240,6 +241,7 @@ StringRef Triple::getArchTypePrefix(ArchType Kind) {
 
   case kalimba:     return "kalimba";
   case lanai:       return "lanai";
+  case dadao:       return "dadao";
   case shave:       return "shave";
   case wasm32:
   case wasm64:      return "wasm";
@@ -507,6 +509,7 @@ Triple::ArchType Triple::getArchTypeForLLVMName(StringRef Name) {
       .Case("spirv64", spirv64)
       .Case("kalimba", kalimba)
       .Case("lanai", lanai)
+      .Case("dadao", dadao)
       .Case("shave", shave)
       .Case("wasm32", wasm32)
       .Case("wasm64", wasm64)
@@ -656,6 +659,7 @@ static Triple::ArchType parseArch(StringRef ArchName) {
                  Triple::spirv64)
           .StartsWith("kalimba", Triple::kalimba)
           .Case("lanai", Triple::lanai)
+          .Case("dadao", Triple::dadao)
           .Case("renderscript32", Triple::renderscript32)
           .Case("renderscript64", Triple::renderscript64)
           .Case("shave", Triple::shave)
@@ -998,6 +1002,7 @@ static Triple::ObjectFormatType getDefaultFormat(const Triple &T) {
   case Triple::hsail:
   case Triple::kalimba:
   case Triple::lanai:
+  case Triple::dadao:
   case Triple::loongarch32:
   case Triple::loongarch64:
   case Triple::m68k:
@@ -1776,6 +1781,7 @@ unsigned Triple::getArchPointerBitWidth(llvm::Triple::ArchType Arch) {
   case llvm::Triple::spirv:
   case llvm::Triple::spir64:
   case llvm::Triple::spirv64:
+  case llvm::Triple::dadao:
   case llvm::Triple::systemz:
   case llvm::Triple::ve:
   case llvm::Triple::wasm64:
@@ -1826,6 +1832,7 @@ Triple Triple::get32BitArchVariant() const {
   case Triple::msp430:
   case Triple::systemz:
   case Triple::ve:
+  case Triple::dadao:
     T.setArch(UnknownArch);
     break;
 
@@ -1944,6 +1951,7 @@ Triple Triple::get64BitArchVariant() const {
   case Triple::ve:
   case Triple::wasm64:
   case Triple::x86_64:
+  case Triple::dadao:
     // Already 64-bit.
     break;
 
@@ -2059,6 +2067,7 @@ Triple Triple::getLittleEndianArchVariant() const {
   switch (getArch()) {
   case Triple::UnknownArch:
   case Triple::lanai:
+  case Triple::dadao:
   case Triple::sparcv9:
   case Triple::systemz:
   case Triple::m68k:
@@ -2358,6 +2367,7 @@ ExceptionHandling Triple::getDefaultExceptionHandling() const {
   case Triple::csky:
   case Triple::hexagon:
   case Triple::lanai:
+  case Triple::dadao:
   case Triple::m68k:
   case Triple::msp430:
   case Triple::systemz:
-- 
2.43.0

